apt/minimization/minimizer.py

```diff
@@ -363,27 +376,90 @@ def fit(self, X: Optional[DATA_PANDAS_NUMPY_TYPE] = None, y: Optional[DATA_PANDA
             # self._cells currently holds the chosen generalization based on target accuracy
 
             # calculate iLoss
-            self._ncp = self._calculate_ncp(X_test, self._generalizations, feature_data)
+            X_test_dataset = ArrayDataset(X_test, features_names=self._features)
+            self.calculate_ncp(X_test_dataset)
 
         # Return the transformer
         return self
 
     def transform(self, X: Optional[DATA_PANDAS_NUMPY_TYPE] = None, features_names: Optional[list] = None,
                   dataset: Optional[ArrayDataset] = None):
-        """ Transforms data records to representative points.
+        """ Transforms data records to representative points. Updates stored ncp value to the one computed on the
```

Comment:
Again, according to the NCP-class suggestion, better something like "Adds a new NCP value" to the NCP score.